src/composables/useUtilities.ts

```diff
@@ -70,20 +71,24 @@ export default function useUtilities() {
     generateDebouncedFn: (delay: number) => (...args: Parameters<F>) => void
   } => {
     let timeout: any
-
-    const wrapDebouncedWithDelay =
-      (delay: number) =>
-        (...args: Parameters<F>) => {
-          clearTimeout(timeout)
-          if (delay > 0) {
-            timeout = setTimeout(() => {
-              fn(...args)
+    const wrapDebouncedWithDelay = (delay: number) => {
```

Comment:
suggestion: can we add a return type?

Reply:
question: Is it necessary since we define the return type of `generateDebouncedFn` directly above?

![image](https://user-images.githubusercontent.com/67973710/236837107-5e950267-7e2a-4dd5-9137-11d6709944c8.png)

Reply:
It's prob fine as is; will infer it